src/libraries/Microsoft.PowerFx.Core/Public/Types/DeferredType.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Types
+{
+    /// <summary>
+    /// An unknown type, attached to syntax nodes whose type cannot be determined.
```

Comment:
consider also commenting that it's not valid to even attempt to execute an expression that references a deferred type